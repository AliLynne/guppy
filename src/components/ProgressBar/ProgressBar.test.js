import React from 'react';
import { render } from 'react-testing-library';
import ProgressBar, { Wrapper } from './ProgressBar';

describe('ProgressBar component', () => {
  describe('should render', () => {
    // we could do the tests in a loop but for three test cases
    // it's OK to duplicate code
    it('with 0%', () => {
      const wrapper = render(<ProgressBar progress={0} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('with 50%', () => {
      const wrapper = render(<ProgressBar progress={0.5} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('with 100%', () => {
      const wrapper = render(<ProgressBar progress={1.0} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('should apply height style to Wrapper', () => {
    const { container, debug } = render(
      <ProgressBar height={10} progress={0.5} />
    );
    //debug(container);
    //expect(container.querySelector('div').getAttribute('height')).toEqual('10');
    //expect(getByTestId('wrapper')).toEqual('test'); //toHaveStyleRule('height', '10px');
  });
});
