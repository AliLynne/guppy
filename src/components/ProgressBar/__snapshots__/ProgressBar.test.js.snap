// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component should render with 0% 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdVaJa cvfKJm"
        height="8"
      >
        <div
          class="sc-bwzfXH ffuotK"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdVaJa cvfKJm"
      height="8"
    >
      <div
        class="sc-bwzfXH ffuotK"
      />
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProgressBar component should render with 50% 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdVaJa cvfKJm"
        height="8"
      >
        <div
          class="sc-bwzfXH ffuotK"
        />
      </div>
    </div>
    <div>
      <div
        class="sc-bdVaJa cvfKJm"
        height="8"
      >
        <div
          class="sc-bwzfXH ffuotK"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdVaJa cvfKJm"
      height="8"
    >
      <div
        class="sc-bwzfXH ffuotK"
      />
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProgressBar component should render with 100% 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdVaJa cvfKJm"
        height="8"
      >
        <div
          class="sc-bwzfXH ffuotK"
        />
      </div>
    </div>
    <div>
      <div
        class="sc-bdVaJa cvfKJm"
        height="8"
      >
        <div
          class="sc-bwzfXH ffuotK"
        />
      </div>
    </div>
    <div>
      <div
        class="sc-bdVaJa cvfKJm"
        height="8"
      >
        <div
          class="sc-bwzfXH ffuotK"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdVaJa cvfKJm"
      height="8"
    >
      <div
        class="sc-bwzfXH ffuotK"
      />
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
